src/layout/rect/iter.rs

```diff
@@ -59,7 +62,9 @@ impl Iterator for Columns {
     ///
     /// Returns `None` when there are no more columns to iterate through.
     fn next(&mut self) -> Option<Self::Item> {
-        if self.current_column >= self.rect.right() {
+        if self.current_column >= self.rect.right()
+            || self.current_column >= self.current_column_back
```

Comment:
given that current column back is initialized to right(), isn't the first check superfluos?
```suggestion
        if self.current_column >= self.current_column_back
```
(and change to `>` if the above minus 1 change is made

Reply:
Good catch, removed.